File: Source/Aladdin.ASN1.ISO/ASN1/ISO/PKCS/PKCS07/DigestedData.cs
using System;

//	DigestedData ::= SEQUENCE {
//		version				INTEGER,
//		digestAlgorithm		AlgorithmIdentifier,
//		encapContentInfo	EncapsulatedContentInfo,
//		digest				OCTET STRING 
//	}

namespace Aladdin.ASN1.ISO.PKCS.PKCS7
{
	public class DigestedData : Sequence
	{
		// информация о структуре
		private static readonly ObjectInfo[] info = new ObjectInfo[] { 

			new ObjectInfo(new ObjectCreator<Integer				>().Factory(), Cast.N), 
			new ObjectInfo(new ObjectCreator<AlgorithmIdentifier	>().Factory(), Cast.N), 
			new ObjectInfo(new ObjectCreator<EncapsulatedContentInfo>().Factory(), Cast.N), 
			new ObjectInfo(new ObjectCreator<OctetString			>().Factory(), Cast.N), 
		}; 
		// конструктор при раскодировании
		public DigestedData(IEncodable encodable) : base(encodable, info) {}

		// конструктор при закодировании
		public DigestedData(Integer version, AlgorithmIdentifier digestAlgorithm, 
			EncapsulatedContentInfo encapContentInfo, OctetString digest) : 
			base(info, version, digestAlgorithm, encapContentInfo, digest) {}

		public Integer						Version				{ get { return (Integer					)this[0]; } } 
		public AlgorithmIdentifier			DigestAlgorithm		{ get { return (AlgorithmIdentifier		)this[1]; } }
		public EncapsulatedContentInfo		EncapContentInfo	{ get { return (EncapsulatedContentInfo	)this[2]; } } 
		public OctetString					Digest				{ get { return (OctetString				)this[3]; } }
	}
}

File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/pkcs/pkcs7/DigestedData.java
package aladdin.asn1.iso.pkcs.pkcs7;
import aladdin.asn1.*; 
import aladdin.asn1.Integer; 
import aladdin.asn1.iso.*; 
import java.io.*;

//	DigestedData ::= SEQUENCE {
//		version				INTEGER,
//		digestAlgorithm		AlgorithmIdentifier,
//		encapContentInfo	EncapsulatedContentInfo,
//		digest				OCTET STRING 
//	}

public final class DigestedData extends Sequence<IEncodable>
{
    private static final long serialVersionUID = -416956041158823784L;
    
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(Integer                .class).factory(), Cast.N), 
		new ObjectInfo(new ObjectCreator(AlgorithmIdentifier	.class).factory(), Cast.N), 
		new ObjectInfo(new ObjectCreator(EncapsulatedContentInfo.class).factory(), Cast.N), 
		new ObjectInfo(new ObjectCreator(OctetString			.class).factory(), Cast.N), 
	}; 
	// конструктор при раскодировании
	public DigestedData(IEncodable encodable) throws IOException { super(encodable, info); }

	// конструктор при закодировании
	public DigestedData(Integer version, AlgorithmIdentifier digestAlgorithm, 
		EncapsulatedContentInfo encapContentInfo, OctetString digest)
	{
		super(info, version, digestAlgorithm, encapContentInfo, digest); 
	}
	public final Integer                    version			() { return (Integer                )get(0); } 
	public final AlgorithmIdentifier		digestAlgorithm	() { return (AlgorithmIdentifier	)get(1); }
	public final EncapsulatedContentInfo	encapContentInfo() { return (EncapsulatedContentInfo)get(2); } 
	public final OctetString                digest			() { return (OctetString            )get(3); }
}
